jena-fuseki2/jena-fuseki-core/src/main/java/org/apache/jena/fuseki/async/AsyncPool.java

```diff
@@ -34,10 +35,16 @@
     // Number of finished tasks kept.
     private static int MAX_FINISHED = 20;
 
-    // See Executors.newCachedThreadPool and Executors.newFixedThreadPool
+    // A ThreadPoolExecutor with
+    // * 0 to nMaxThreads
+    // * no queue of waiting tasks (tasks execute or are rejected)
+    // * dormant threads released after 120s.
+    //
+    // SynchronousQueue is a BlockingQueue that has zero length - it accepts and
+    // delivers an item or rejects immediately, no delay by queueing.
```

Comment:
+1 So provided there is at least one thread free then submitting a task will succeed, if all threads are in-use then it fails

Reply:
Yes - like that.  The actual mechanism is expands up to nMaxThreads, creating threads if needed.